src/stores/certificate-trust-api.ts

```diff
@@ -0,0 +1,29 @@
+import { createApi, fetchBaseQuery, FetchBaseQueryError } from '@reduxjs/toolkit/query/react';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+
+export const certificateTrustApi = createApi( {
+	reducerPath: 'certificateTrustApi',
+	baseQuery: fetchBaseQuery(),
+	tagTypes: [ 'CertificateTrust' ],
+	refetchOnFocus: true,
+	endpoints: ( builder ) => ( {
+		checkCertificateTrust: builder.query< boolean, void >( {
+			queryFn: async () => {
+				try {
+					const isTrusted = await getIpcApi().isCATrusted();
+					return { data: isTrusted };
+				} catch ( error ) {
+					console.error( 'Failed to check certificate trust:', error );
+					const errorMessage =
+						error instanceof Error ? error.message : 'Failed to check certificate trust';
+					return {
+						error: { data: false, error: errorMessage } as FetchBaseQueryError,
+					};
```

Comment:
```suggestion
					return {
						error: { status: 500, data: errorMessage },
					};
```
This seems to match the `FetchBaseQueryError` type.

Reply:
Thanks! Updated in bd34055cec87fff47b4a13adf8da74bc3a8dec42!